packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -259,3 +260,157 @@ export function addUrlProtocolPrefix(urlString: string): string {
 export function ethersBigNumberToBN(bigNumber: BigNumber): BN {
   return new BN(stripHexPrefix(bigNumber.toHexString()), 'hex');
 }
+
+/**
+ * Returns the total fiat value of a given account on a given network,
+ * including native currency and token values
+ *
+ * @param CurrencyRateController - The CurrencyRateController instance
+ * @param PreferencesController - The PreferencesController instance
+ * @param AccountTrackerController - The AccountTrackerController instance
+ * @param TokenBalancesController - The TokenBalancesController instance
+ * @param TokenRatesController - The TokenRatesController instance
+ * @param TokensController - The TokensController instance
+ * @returns A number representing the total fiat balance of an account on a network.
+ */
+export function getTotalFiatAccountBalance(
+  CurrencyRateController,
```

Comment:
Controllers as parameters is something we've been trying to move away from as a pattern because it makes it difficult to tell which methods are used where. Could you update this to pass in state values instead? I know that does mean we'll need to repeat the "state retrieval" steps between clients, but I think that would still be an easier pattern for us to maintain to start with.

Reply:
I was thinking just this same thing!  Glad we're aligned!